### PR TITLE
Skip redis tests until PHP7.2 stabilizes

### PR DESCRIPTION
**Pull request by markstory**
This will let us get passing builds out of PHP7.2 and hopefully make changes in future beta's more visible.

### CODE REVIEW

**Comment by ADmad** [comment]
Seems the check needs to be tweaked, redis tests were still run on 7.2

File: tests/TestCase/Cache/Engine/RedisEngineTest.php [hunk 1]
```diff
@@ -33,6 +33,7 @@ public function setUp()
     {
         parent::setUp();
         $this->skipIf(!class_exists('Redis'), 'Redis extension is not installed or configured properly.');
+        $this->skipIf(version_compare(PHP_VERSION, '7.2.0beta', '>='), 'Redis is misbehaving in PHP7.2');
```
Comment (antograssiot) [inline tests/TestCase/Cache/Engine/RedisEngineTest.php]: @markstory the version used is 7.2.0-dev see https://travis-ci.org/cakephp/cakephp/jobs/262358142#L505

Comment (markstory) [inline tests/TestCase/Cache/Engine/RedisEngineTest.php]: I'll get that updated.

**Comment by ADmad** [comment]
A new issue has cropped up :( https://travis-ci.org/cakephp/cakephp/jobs/265292163#L658

**Comment by markstory** [comment]
And then there is this [lolz build](https://travis-ci.org/cakephp/cakephp/jobs/265292165)

File: src/Collection/Iterator/SortIterator.php [hunk 2]
```diff
@@ -65,12 +65,12 @@ public function __construct($items, $callback, $dir = SORT_DESC, $type = SORT_NU
...
         $callback = $this->_propertyExtractor($callback);
         $results = [];
-        foreach ($items as $key => $value) {
-            $value = $callback($value);
-            if ($value instanceof DateTimeInterface && $type === SORT_NUMERIC) {
-                $value = $value->format('U');
+        foreach ($items as $key => $val) {
+            $val = $callback($val);
+            if ($val instanceof DateTimeInterface && $type === SORT_NUMERIC) {
+                $val = $value->format('U');
```
Comment (ADmad) [inline src/Collection/Iterator/SortIterator.php]: You missed renaming one instance of $value here.

Comment (markstory) [reply src/Collection/Iterator/SortIterator.php]: Whoops.

**Comment by markstory** [comment]
Are folks ok with the variable renames? They seemed to have tricked the tests into passing!

**Review by ADmad — APPROVED** [review]

**Pull request merged**